app/forms/questionnaire_form.py

```diff
@@ -225,9 +234,19 @@ def _validate_calculated_question(
         return True
 
     def _get_period_range_for_single_date(self, date_from, date_to):
-        handler = DateHandler(
-            date_from, {}, self.answer_store, self.metadata, location=self.location
+
+        list_item_id = self.location.list_item_id if self.location else None
+        value_source_resolver = ValueSourceResolver(
+            answer_store=self.answer_store,
+            list_store=self.list_store,
+            metadata=self.metadata,
+            schema=self.schema,
+            location=self.location,
+            list_item_id=list_item_id,
+            escape_answer_values=False,
         )
```

Comment:
This and these same lines in `get_answer_fields` could be removed if `value_source_resolver` was a property of the `QuestionnaireForm`. Not sure if that is necessarily the right place or whether it should in block handler, but either way I think one of those approaches is more appropriate than initialising inline here and in `get_answer_fields`.

Reply:
As discussed, `get_answer_fields` doesn't have access to the form yet, so this can't be done for that. It could be done for the date handler but since there is only a single usage at the moment, it decided it was okay to keep as is for now.